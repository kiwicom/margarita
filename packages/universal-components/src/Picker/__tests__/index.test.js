// @flow

import * as React from 'react';
import { render, shallow } from 'react-native-testing-library';

import { Picker } from '..';

describe('Picker', () => {
  const testData = [{ label: 'A', value: 'a' }, { label: 'B', value: 'b' }];
  const onValueChange = jest.fn();
  const confirmLabel = 'OK';

  const { getByText } = render(
    <Picker
      optionsData={testData}
      selectedValue={testData[1].value}
      onValueChange={onValueChange}
      confirmLabel={confirmLabel}
    />,
  );

  it('should contain correct label', () => {
    expect(getByText(testData[1].label)).toBeDefined();
  });

  it('renders', () => {
    expect(
      shallow(
        <Picker
          optionsData={testData}
          selectedValue={testData[0].value}
          onValueChange={jest.fn()}
          confirmLabel={confirmLabel}
        />,
      ),
    ).toMatchSnapshot();
  });
});
