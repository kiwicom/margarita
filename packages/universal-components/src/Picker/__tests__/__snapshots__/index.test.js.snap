// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker renders 1`] = `
Object {
  "output": <Component>
    <Component>
      <Button
        label="A"
        onPress={[Function]}
        rightIcon={
          <Icon
            name="chevron-right"
          />
        }
        type="secondary"
      />
      <Modal
        backdropOpacity={0.5}
        isVisible={false}
        onBackdropPress={[Function]}
        onRequestClose={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
            "margin": 0,
          }
        }
      >
        <Component
          style={
            Object {
              "backgroundColor": "#fff",
              "borderTopEndRadius": 6,
              "borderTopStartRadius": 6,
              "paddingBottom": 40,
              "paddingTop": 4,
              "width": "100%",
            }
          }
        >
          <NativePicker
            onValueChange={[Function]}
            optionsData={
              Array [
                Object {
                  "label": "A",
                  "value": "a",
                },
                Object {
                  "label": "B",
                  "value": "b",
                },
              ]
            }
            selectedValue="a"
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <Component
            style={
              Object {
                "marginHorizontal": 8,
              }
            }
          >
            <Button
              label="OK"
              onPress={[Function]}
              type="secondary"
            />
          </Component>
        </Component>
      </Modal>
    </Component>
  </Component>,
}
`;
